core/container.go

```diff
@@ -288,6 +379,21 @@ func (mnt ContainerMount) SourceState() (llb.State, error) {
 	return defToState(mnt.Source)
 }
 
+var _ HasRawDigest = (*ContainerMount)(nil)
+
+func (mnt ContainerMount) RawDigest() digest.Digest {
+	return dagql.HashFrom(
+		"containerMount",
+		mnt.SourcePath,
+		mnt.Target,
```

Comment:
nit: this seems like a sneaky source of collisions - say in one call I mount `/foo/bar` to `/baz`, and in another I mount `/foo` to `/bar/baz`. Those will both concatenate the same. A bit contrived, but still gives me the heebie-jeebies.

Maybe we could change `HashFrom` to write a null byte after every value as a delineator?

Reply:
Yeah definitely! I thought about doing the null byte thing, but I guess you could still pass null bytes into it?

Maybe we could escape null bytes before injecting them? Don't really know, will have a think!

Reply:
Two things:
- I've changed `HashFrom` to put in null bytes.
- I've removed the hellish `RawDigest` thing, and instead just made a generic `DigestOf` function (seems similar to what @sipsma is doing in #10497). And then have a special `Container.WithoutInputs` to "pureify" the container from the buildkit stuffs.

So these should both make sure we avoid nasty collisions.